desc/magnetic_fields.py

```diff
@@ -478,6 +478,12 @@ def __init__(self, *fields):
             [type(field) for field in fields]
         )
         self._fields = fields
+        offset = jnp.concatenate(
+            [jnp.array([0]), jnp.cumsum(jnp.array([s.dim_x for s in self]))[:-1]]
+        )
+        # store split indices for unpacking
+        # as a numpy array to avoid jax issues later
+        self._split_idx = np.asarray(offset)
```

Comment:
Way to put this in a super init for `OptimizableCollection`? otherwise this would have to be done in every init of subclasses of `OptimizableCollection`

Reply:
what is this needed for? I think there already is something similar in `OptimizableCollection`

Reply:
There was, but it used `jnp.split` and jax threw an error complaining that it could not deal with non-static indices, and it seems that [`jnp.split` cannot be jitted, but if you use an np array as the indices it works ](https://github.com/google/jax/issues/13769#issuecomment-1496709173)

Reply:
hmmm ok. We could add an init method to do this, and then call super init from all the subclasses. It might also be possible to do some metaclass stuff like we do to register things as pytrees:
https://github.com/PlasmaControl/DESC/blob/969774a4ef082689bf6a9f1a222be6e3f3a47d34/desc/io/equilibrium_io.py#L107

 Another option might be something like we do in the `Optimizable` class where it gets built the first time its called:
https://github.com/PlasmaControl/DESC/blob/969774a4ef082689bf6a9f1a222be6e3f3a47d34/desc/optimizable.py#L18